discovery-provider/src/utils/session_manager.py

```diff
@@ -11,10 +14,24 @@
 class SessionManager:
     def __init__(self, db_url, db_engine_args):
         self._engine = create_engine(db_url, **db_engine_args)
+
```

Comment:
These events need some comments and documentation, maybe some links

Reply:
Added comments and links to documentation.

Reply:
amazing! i think i understand now. so when a tx begins, the scoped_session function attaches the name of the calling function to the session.

the after begin takes the name from the session and writes it to the connection.

the before_cursor_execute appends the connection src param to the statement. is this right? 

the before_cursor_execute takes the name from the session and writes it to the connection

Reply:
yup!